clients/application-amqp/src/test/java/org/eclipse/hono/application/client/amqp/ProtonBasedApplicationClientTest.java

```diff
@@ -75,7 +77,9 @@
     @BeforeEach
     void setUp() {
         final var vertx = mock(Vertx.class);
+        when(vertx.eventBus()).thenReturn(mock(EventBus.class));
```

Comment:
what do we need the event bus for?

Reply:
`SenderCachingServiceClient` invokes the below piece of code if `isTenantSpecificLink` is true. Hence the event bus is needed in the test.
```
        if (isTenantSpecificLink) {
            connection.getVertx().eventBus().consumer(Constants.EVENT_BUS_ADDRESS_TENANT_TIMED_OUT,
                    this::handleTenantTimeout);
        }
```